src/lcm/simulation/simulate.py

```diff
@@ -1,218 +1,220 @@
-import inspect
 import logging
 from collections.abc import Callable
 from functools import partial
-from typing import Any
 
 import jax
 import jax.numpy as jnp
 import pandas as pd
-from dags import concatenate_functions
 from jax import Array, vmap
 
-from lcm.argmax import argmax
-from lcm.dispatchers import spacemap, vmap_1d
-from lcm.interfaces import InternalModel, StateChoiceSpace
-from lcm.typing import InternalUserFunction, ParamsDict
+from lcm.discrete_problem import get_solve_discrete_problem_policy
+from lcm.dispatchers import simulation_spacemap, vmap_1d
+from lcm.interfaces import (
+    InternalModel,
+    InternalSimulationPeriodResults,
+    StateChoiceSpace,
+)
+from lcm.random import draw_random_seed, generate_simulation_keys
+from lcm.simulation.processing import as_data_frame, process_simulated_data
+from lcm.state_choice_space import create_state_choice_space
+from lcm.typing import ParamsDict
+
+
+def solve_and_simulate(
+    params: ParamsDict,
+    initial_states: dict[str, Array],
+    compute_ccv_policy_functions: dict[int, Callable[..., tuple[Array, Array]]],
+    model: InternalModel,
+    next_state: Callable[..., dict[str, Array]],
+    logger: logging.Logger,
+    solve_model: Callable[..., dict[int, Array]],
+    *,
+    additional_targets: list[str] | None = None,
+    seed: int | None = None,
+) -> pd.DataFrame:
+    """First solve the model and then simulate the model forward in time.
+
+    Same docstring as `simulate` mutatis mutandis.
+
+    """
+    vf_arr_dict = solve_model(params)
+    return simulate(
+        params=params,
+        initial_states=initial_states,
+        compute_ccv_policy_functions=compute_ccv_policy_functions,
+        model=model,
+        next_state=next_state,
+        logger=logger,
+        vf_arr_dict=vf_arr_dict,
+        additional_targets=additional_targets,
+        seed=seed,
+    )
 
 
 def simulate(
     params: ParamsDict,
     initial_states: dict[str, Array],
```

Comment:
I did not try to understand in all detail, but at least `as_data_frame` makes a balanced panel out of this. We must eventually allow starting from different ages, though. Maybe make an issue out of this? *(while looking for whether we have one already, I added the fix for #96, example needed updating anyhow)*